GeosCore/fullchem_mod.F90

```diff
@@ -1375,6 +1407,13 @@ SUBROUTINE Do_FullChem( Input_Opt,  State_Chm, State_Diag,                 &
           ! Skip if this is not a GEOS-Chem species
           IF ( SpcID <= 0 ) CYCLE
 
+          ! Scan for negatives
+          IF ( State_Diag%Archive_KppNegatives ) THEN
+             IF ( C(N) < 0.0_dp ) THEN 
+                State_Diag%KppNegatives(I,J,L) = State_Diag%KppNegatives(I,J,L) + 1.0_dp 
```

Comment:
The increment should be `1.0_f4` (since the array is REAL(f4), or single precision.  Fortran will cast it anyway, but I think using the proper KIND value prevents the cast.